src/GraphQL.Tests/Validation/ValidationSchema.cs

```diff
@@ -47,7 +47,7 @@ public class Dog : ObjectGraphType
     public Dog()
     {
         Field<StringGraphType>("name")
-            .Argument<StringGraphType>("surname");
+            .Argument<BooleanGraphType>("surname");
```

Comment:
Interesting how many mistakes like this we have in our schema 🥴
Must create an analyzer for interfaces. Otherwise, upgrading to v8 may be a hell...

Reply:
I’m not so concerned. If someone has a bug of this nature, it’s critically breaking their schema. The exceptions they will receive will describe the location of the error, and I would expect they would be happy to know about it. 

Also FYI, our code is slightly more lenient than the spec. It allows non-nullable args to be used for a nullable arg on an implemented field. (The spec says “invariant”.)

Reply:
What is the behavior in v7 when you have such a bug in the schema? Will it fail the query? If it fails and users discover the mistake only during the upgrade to v8, it means the argument is not really used by the clients, so it's not a problem to fix the schema. However, if it works in v7 and stops working in v8, it requires fixing clients before upgrading to v8.
It would be great to add this validation to v7 and write it into logs as a warning to allow users to fix their schema before the upgrade, but unfortunately, we don't have a logger dependency. Then, an analyzer may be the best option for an easier upgrade. Report it as a warning in v7 and an error in v8. But the urgency depends on the answer to the question whether it fails the query or not.

Reply:
It will now fail initialization.  People can remove schema initialization checks if need be; execution will be the same.  So for example if the field had a non-null argument but the interface didn't, clients could execute a request on the interface via a fragment, which will cause the field resolver to execute without the required argument -- same as v7.

For reference, always in the past when we found similar missing validation checks, which would cause a breaking bug in the expected operation in certain cases, we have added the validation checks without hesitation, even sometimes on a minor version.  Basically, we consider such findings as bugs within GraphQL.NET, eligible for immediately fixing.  Obviously we prefer to make such changes on a major version.

Reply:
This perhaps would be quite valuable if many schema initialization errors are expected:
- #3875 

Do you think we should consider adding for v8?

Reply:
> So for example if the field had a non-null argument but the interface didn't

But what about the wrong field type? I assume it's the same, right?

Reply:
It would execute the same as before.  In the situation here, it would read the boolean, and then fail during the call to `.GetArgument<string>`, I think, or however the `ValueConverter` is configured to deal with that conversion.